services/consensuscontext/validate_transactions_block.go

```diff
@@ -86,27 +88,33 @@ func validateTransactionsBlockMetadataHash(ctx context.Context, vctx *txValidato
 	})
 }
 
-func validateTxTransactionOrdering(ctx context.Context, vctx *txValidatorContext) error {
+func (s *service) validateTxTransactionOrdering(ctx context.Context, transactionBlock *protocol.TransactionsBlockContainer) error {
+	txs := transactionBlock.SignedTransactions
+	if s.config.ConsensusContextTriggersEnabled() {
+		if len(txs) == 0 {
+			return ErrTriggerEnabledAndTriggerMissing
+		}
+		txs = txs[:len(txs)-1]
```

Comment:
I might be missing it, but what if the last tx isn’t a trigger here?

Reply:
order of validations ... before the ordering we validate trigger compliance. so if it passed then the last one is a trigger tx.